lib/config/configuration.go

```diff
@@ -451,6 +440,65 @@ func ApplyFileConfig(fc *FileConfig, cfg *service.Config) error {
 	return nil
 }
 
+func applyAuthOrProxyAddress(fc *FileConfig, cfg *service.Config) error {
+	switch cfg.Version {
+	// For config versions v1 and v2, the auth_servers field can point to an auth
+	// server or a proxy server
+	case defaults.TeleportConfigVersionV1, defaults.TeleportConfigVersionV2:
+		// config file has auth servers in there?
+		if len(fc.AuthServers) > 0 {
+			var parsedAddresses []utils.NetAddr
+
+			for _, as := range fc.AuthServers {
+				addr, err := utils.ParseHostPortAddr(as, defaults.AuthListenPort)
+				if err != nil {
+					return trace.Wrap(err)
+				}
+
+				parsedAddresses = append(parsedAddresses, *addr)
+			}
+
+			if err := cfg.SetAuthServerAddresses(parsedAddresses); err != nil {
+				return trace.Wrap(err)
+			}
+		}
+
+		if fc.AuthServer != "" {
+			return trace.BadParameter("auth_server is supported from config version v3 onwards")
+		}
+
+		if fc.ProxyServer != "" {
+			return trace.BadParameter("proxy_server is supported from config version v3 onwards")
+		}
+
+	// From v3 onwards, either auth_server or proxy_server should be set
+	case defaults.TeleportConfigVersionV3:
+		if len(fc.AuthServers) > 0 {
+			return trace.BadParameter("config v3 has replaced auth_servers with either auth_server or proxy_server")
+		}
+
+		if fc.AuthServer != "" {
```

Comment:
I had avoided it here because there is error checking for this that happens when `Config` is validated.

But yeah, I agree we should return an error here too. It saves extra work and makes us exit early if the file config is not valid. Plus it makes sense that it should be checked here also, as it's the `FileConfig` spec that disallows them to be set at the same time in the first place.